huggingface/pytorch/training/docker/1.11/py3/cu113/Dockerfile.trcomp.gpu

```diff
@@ -0,0 +1,113 @@
+ && apt install -y libomp5 \
+ && apt-get clean \  
+ && rm -rf /var/lib/apt/lists/*
+
+
+# Install PyTorch
+RUN pip uninstall -y torch \
+ && pip install --no-deps --no-cache-dir -U --force-reinstall ${PT_URL} \
+ && rm -rf ${PT_URL}
+
+# Install PyTorch XLA
+RUN pip uninstall -y torch_xla \
+ && pip install --no-deps --no-cache-dir -U --force-reinstall ${PT_XLA_URL} \
+ && rm -rf ${PT_XLA_URL}
+
+# Install TorchVision
+RUN pip uninstall -y torchvision \
+ && pip install --no-deps --no-cache-dir -U --force-reinstall ${TORCHVISION_URL} \
+ && rm -rf ${TORCHVISION_URL}
+
+# Install TorchAudio
+RUN pip uninstall -y torchaudio \
+ && pip install --no-deps --no-cache-dir -U --force-reinstall ${TORCHAUDIO_URL} \
+ && rm -rf ${TORCHAUDIO_URL}
+
+# Install Transformers
+RUN pip uninstall -y transformers \
+ && pip install --no-deps --no-cache-dir --force-reinstall -U ${HF_TRANSFORMERS_URL} \
+ && rm -rf ${HF_TRANSFORMERS_URL}
+
+# Install SageMaker Training Toolkit
+RUN pip uninstall -y sagemaker-training \
+ && pip install --no-deps --no-cache-dir -U --force-reinstall "sagemaker-training>=4.2.6"
+
+# Install SageMaker PyTorch Training Toolkit
+RUN pip uninstall -y sagemaker-pytorch-training \
+ && pip install --no-deps --no-cache-dir -U --force-reinstall "sagemaker-pytorch-training>=2.6.1,<3"
+
+
+ARG CONDA_PREFIX=/opt/conda
+
+# Install common training dependencies
+RUN ${CONDA_PREFIX}/bin/conda install -y -c conda-forge librosa
+RUN pip install --no-cache-dir \
+    sacrebleu \
+    fugashi \
+    ipadic \
+    nltk \
+    rouge-score \
+    soundfile \
+    sentence-transformers
+
+# Numpy version gets downgraded and becomes incompatible with PyTorch wheel in the above conda command.
+RUN pip install -U numpy==1.22.2
+
+# Fix library links
+RUN ln -s ${CONDA_PREFIX}/lib/libmkl_intel_lp64.so ${CONDA_PREFIX}/lib/libmkl_intel_lp64.so.1 \
+ && ln -s ${CONDA_PREFIX}/lib/libmkl_intel_thread.so ${CONDA_PREFIX}/lib/libmkl_intel_thread.so.1 \
+ && ln -s ${CONDA_PREFIX}/lib/libmkl_core.so ${CONDA_PREFIX}/lib/libmkl_core.so.1 \
+ && ln -s ${CONDA_PREFIX}/lib/libmkl_gnu_thread.so ${CONDA_PREFIX}/lib/libmkl_gnu_thread.so.1
+
+# Install Horovod
+ENV HOROVOD_VERSION=0.24.3
+RUN pip uninstall -y horovod \
+ && ldconfig /usr/local/cuda-11.3/targets/x86_64-linux/lib/stubs \
+ && HOROVOD_GPU_ALLREDUCE=NCCL HOROVOD_CUDA_HOME=/usr/local/cuda-11.3 HOROVOD_WITH_PYTORCH=1 pip install --no-cache-dir horovod==${HOROVOD_VERSION} \
+ && ldconfig
+
+RUN HOME_DIR=/root \
+ && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
+ && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
+ && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
+ && chmod +x /usr/local/bin/testOSSCompliance \
+ && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
+ && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
+ && rm -rf ${HOME_DIR}/oss_compliance*
```

Comment:
For standard DLCs, we add license like: https://github.com/aws/deep-learning-containers/blob/master/pytorch/training/docker/1.11/py3/cu113/Dockerfile.gpu#L220 . That license should be present in the base. Do we need any other license for Hopper or is the one in the base image fine?